### DISCUSSION TITLE
Folder structure view

### DISCUSSION THREAD

**Discussion by will-stone** [alefragnani/vscode-project-manager#562] [Q&A] [answered]
Hi,

Is it possible to display the projects as a tree of folders in the sidebar? I organise my projects by service and it would be nice to have that automatically reflected. e.g.

```
/Documents
├── github/
│   ├── project-1
│   └── project-2
├── gitlab/
│   ├── project-a
│   └── project-b
└── srht/
    ├── project-x
    ├── project-y
    └── project-z
```

Thanks 🙂

**Comment by alefragnani** [comment, marked as answer]
Hi @will-stone ,

No, it is not possible. 

The Side Bar is not designed to reflect a file system structure but instead, list your projects (saved/favorites or detected). 

But, in recent releases you can organize your saved/favorite projects by **Tags** (https://github.com/alefragnani/vscode-project-manager#project-tags---view-and-filter ). It’s simple and fairly easy to use. I suggest you to take a look if it fit your needs.

Hope this helps

**Reply by will-stone** [reply]
Thanks @alefragnani I have tried the favourite system, but I feel that the file system _is_ my style of tagging and would rather not organise twice. No worries, just thought I’d ask in case I was missing something. Thanks!

**Reply by alefragnani** [reply]
Feel free to open a Feature Request, so other users that share the same requirements can iterate/upvote. 

Depending on the feedback, it could be added to the backlog

**Reply by will-stone** [reply]
Ah, okay. Will do. Thanks 😊

**Comment by mohitsinghs** [comment]
Hey @will-stone, I know I'm little late but I am on the same boat and like you, I also tried favorite system. While I love what @alefragnani have created, I ended up creating [another extenion](https://github.com/mohitsinghs/vscode-repo) based on Project Manager that uses a rust binary to do the lookup part and arranges everything in a tree view according to folders.

**Reply by will-stone** [reply]
Very nice! I would feel bad cheating on Project Manager but your project certainly looks enticing, I'll give it a go shortly.

**Reply by alefragnani** [reply]
@will-stone no problem! We must use/combine the extensions that better fit our needs. My extension is aimed to simplify the visibility of projects only, _hiding_ the folder structure. For those that prefers it visible, I guess @mohitsinghs extension may be a good fit.

@mohitsinghs congrats for your extension!

**Reply by mohitsinghs** [reply]
Thanks @alefragnani . It's still not mature though, specially on platforms that I don't have access to, like macOS. In fact @will-stone just [triggered a bug](https://github.com/mohitsinghs/vscode-repo/issues/5). While moving repository detection to rust made it instantaneous, the side effect is platform support.

**Reply by alefragnani** [reply]
Yeah, that's a side effect. 

You can however, limit your extension to be _platform specific_ so it won't be available on unsupported platforms until you intentionally enable it. Should avoid issues that you are not able to deal with, at the moment. I've faced similar issues with other extensions in the past, when there were no official _platform specific_ support.

Hope this helps
